app/Actions/Albums/PositionData.php

```diff
@@ -2,37 +2,37 @@
 
 namespace App\Actions\Albums;
 
-use App\Actions\Album\Extensions\LocationData;
-use App\Actions\Albums\Extensions\PublicIds;
+use App\Actions\PhotoAuthorisationProvider;
 use App\Models\Configs;
 use App\Models\Photo;
 
 class PositionData
 {
-	use LocationData;
+	protected PhotoAuthorisationProvider $photoAuthorisationProvider;
+
+	public function __construct()
+	{
+		$this->photoAuthorisationProvider = resolve(PhotoAuthorisationProvider::class);
+		// caching to avoid further request
+		Configs::get();
+	}
 
 	/**
 	 * Given a list of albums, generate an array to be returned.
 	 *
-	 * @param Collection[Album] $albums
-	 *
 	 * @return array
 	 */
-	public function do()
+	public function do(): array
 	{
-		// caching to avoid further request
-		Configs::get();
-
-		// Initialize return var
-		$return = [];
-
-		$albumIDs = resolve(PublicIds::class)->getPublicAlbumsId();
-
-		$query = Photo::with('album')->whereIn('album_id', $albumIDs);
-
-		$full_photo = Configs::get_value('full_photo', '1') == '1';
-		$return['photos'] = $this->photosLocationData($query, $full_photo);
-
-		return $return;
+		$result = [];
+		$result['id'] = null;
+		$result['title'] = null;
+		$result['photos'] = $this->photoAuthorisationProvider->applySearchabilityFilter(
+			Photo::with(['album', 'size_variants_raw', 'size_variants_raw.sym_links'])
+				->whereNotNull('latitude')
+				->whereNotNull('longitude')
+		)->get()->toArray();
```

Comment:
I'm wondering if we should keep this one as a Collection `->get()` or as an array `->get()->toArray()`.
I am more in favour of the first.

Reply:
The idea was to keep it consistent with the other return types in the same folder, i.e. with the actions `Smart`, `Top` and `Tree`. If you look at `AlbumsController` you see that each of the four actions returns an array.